feat: overhaul RPC server

zivkovicmilos opened the pull request:
## Description

Chips away at https://github.com/gnolang/gno/issues/4896.

This PR overhauls the underlying RPC server implementation in Tendermint2, to optimize on request / response processing. It significantly reduces the code complexity, and general setup overhead.

To the user, the interaction with the node's RPC should stay the same, it's just the internal processing that is changed.

I've modularized different parts of the RPC handling process, so we eliminated globals. I've also added testing suites for each component, which should be easy to expand in the future.

## A note for reviewers

Don't be intimidated by the line changes / files changed.
Most of the files affected are simple renames.

There is no point in viewing the old RPC implementation for comparison as this PR completely removes it, apart from one aspect, which is **to see if we broke backwards compatibility**, especially when it comes to dealing with Amino.

For this, I'd suggest comparing request processing flows for the old RPC implementation on `master`, and on this branch. I've generated some Mermaid diagrams below to make it easier to follow along.

## Overview

```mermaid
  graph TB
      subgraph "Client Layer"
          HTTP_CLIENT["HTTP Client"]
          WS_CLIENT["WebSocket Client"]
      end

      subgraph "Server Entry (core/server.go)"
          SERVER["Server<br/>• Start/Stop lifecycle<br/>• TCP/UNIX binding<br/>• Graceful shutdown"]
      end

      subgraph "Protocol Layer (lib/server/jsonrpc.go)"
          JSONRPC["JSONRPC Handler<br/>• Route registration<br/>• Request parsing<br/>• Response writing"]

          subgraph "Request Handlers"
              HTTP_POST["POST /<br/>JSON-RPC body"]
              HTTP_GET["GET /{method}<br/>Query params"]
              WS_HANDLER["WS /websocket<br/>Melody sessions"]
              INDEX["GET /<br/>Endpoint index"]
          end

          subgraph "Middleware"
              CONTENT["Content-Type<br/>application/json"]
              SIZE["Max Size<br/>1MB limit"]
          end
      end

      subgraph "Handler Registry (core/register.go)"
          REGISTRY["Setup Functions"]

          ABCI["SetupABCI<br/>• abci_info<br/>• abci_query"]
          BLOCKS["SetupBlocks<br/>• blockchain<br/>• block<br/>• commit<br/>• block_results"]
          CONSENSUS["SetupConsensus<br/>• validators<br/>• consensus_state<br/>• consensus_params"]
          MEMPOOL["SetupMempool<br/>• broadcast_tx_*<br/>• unconfirmed_txs"]
          NET["SetupNet<br/>• net_info<br/>• genesis"]
          STATUS["SetupStatus<br/>• status"]
          TX["SetupTx<br/>• tx"]
          HEALTH["SetupHealth<br/>• health"]
      end

      subgraph "Node Dependencies"
          BLOCKSTORE["BlockStore"]
          STATEDB["StateDB"]
          MEMPOOL_DEP["Mempool"]
          CONSENSUS_DEP["ConsensusState"]
          P2P["P2P/Peers"]
      end

      HTTP_CLIENT --> SERVER
      WS_CLIENT --> SERVER
      SERVER --> JSONRPC

      JSONRPC --> HTTP_POST
      JSONRPC --> HTTP_GET
      JSONRPC --> WS_HANDLER
      JSONRPC --> INDEX

      HTTP_POST --> CONTENT --> SIZE
      HTTP_GET --> CONTENT --> SIZE

      JSONRPC --> REGISTRY
      REGISTRY --> ABCI
      REGISTRY --> BLOCKS
      REGISTRY --> CONSENSUS
      REGISTRY --> MEMPOOL
      REGISTRY --> NET
      REGISTRY --> STATUS
      REGISTRY --> TX
      REGISTRY --> HEALTH

      BLOCKS --> BLOCKSTORE
      BLOCKS --> STATEDB
      TX --> BLOCKSTORE
      TX --> STATEDB
      MEMPOOL --> MEMPOOL_DEP
      CONSENSUS --> CONSENSUS_DEP
      CONSENSUS --> P2P
      NET --> P2P

```

## Request flow (HTTP)

```mermaid
  sequenceDiagram
      participant C as Client
      participant S as Server
      participant J as JSONRPC
      participant R as Router
      participant H as Handler
      participant N as Node

      C->>S: HTTP POST / (JSON-RPC 2.0)
      S->>J: handleHTTPRequest
      J->>J: extractBaseRequests<br/>(single or batch)
      J->>J: isValidBaseRequest<br/>(validate jsonrpc: "2.0")

      loop For each request
          J->>R: route(metadata, request)
          R->>R: Lookup handler by method
          R->>H: handler(metadata, params)
          H->>N: Query node state
          N-->>H: Result
          H-->>R: (result, error)
          R-->>J: BaseJSONResponse
      end

      J->>J: Build response<br/>(single or batch array)
      J-->>S: WriteResponse
      S-->>C: JSON Response

```

## WS flow

```mermaid
  sequenceDiagram
      participant C as Client
      participant S as Server
      participant M as Melody
      participant CM as ConnManager

      C->>S: GET /websocket (Upgrade)
      S->>M: HandleRequest
      M->>M: HandleConnect
      M->>CM: AddWSConnection(uuid)
      M-->>C: Connection established

      loop Messages
          C->>M: JSON-RPC message
          M->>S: HandleMessage(session, msg)
          S->>S: Process request
          S->>M: WriteResponse(session)
          M-->>C: JSON-RPC response
      end

      C->>M: Close connection
      M->>M: HandleDisconnect
      M->>CM: RemoveWSConnection(uuid)

```

zivkovicmilos committed:
Start refactoring the RPC server

Gno2D2 commented:
#### 🛠 PR Checks Summary
All **Automated Checks** passed. ✅

##### Manual Checks (for Reviewers):
- [ ] **IGNORE** the bot requirements for this PR (force green CI check)


<details><summary>Read More</summary>

🤖 This bot helps streamline PR reviews by verifying automated checks and providing guidance for contributors and reviewers.

##### ✅ Automated Checks (for Contributors):
*No automated checks match this pull request.*

##### ☑️ Contributor Actions:
1. Fix any issues flagged by automated checks.
2. Follow the Contributor Checklist to ensure your PR is ready for review.
	- Add new tests, or document why they are unnecessary.
	- Provide clear examples/screenshots, if necessary.
	- Update documentation, if required.
	- Ensure no breaking changes, or include `BREAKING CHANGE` notes.
	- Link related issues/PRs, where applicable.

##### ☑️ Reviewer Actions:
1. Complete manual checks for the PR, including the guidelines and additional checks if applicable.

##### 📚 Resources:
- [Report a bug with the bot](https://www.github.com/gnolang/gno/issues/3238).
- [View the bot’s configuration file](https://github.com/gnolang/gno/tree/master/contribs/github-bot/internal/config/config.go).

<details><summary><b>Debug</b></summary><blockquote>


<details><summary><b>Manual Checks</b></summary><blockquote>

<details><summary>**IGNORE** the bot requirements for this PR (force green CI check)</summary><blockquote>

### If
```
🟢 Condition met
└── 🟢 On every pull request

```
### Can be checked by

- Any user with comment edit permission

</blockquote></details>

</blockquote></details>

</blockquote></details>

</details>

zivkovicmilos committed:
Add support for the route index page, GET handling for each route

zivkovicmilos committed:
Add abci, blocks methods

zivkovicmilos committed:
Add block RPC tests

zivkovicmilos committed:
Migrate legacy range test, drop legacy handlers

zivkovicmilos committed:
Add consensus RPC endpoints

zivkovicmilos committed:
Drop unused endpoints

zivkovicmilos committed:
Add health RPC endpoint

zivkovicmilos committed:
Remove redundant doc comments

zivkovicmilos committed:
Add mempool RPC endpoints

zivkovicmilos committed:
Add net RPC endpoints

zivkovicmilos committed:
Add status RPC endpoint

zivkovicmilos committed:
Add tx RPC endpoint

zivkovicmilos committed:
Remove unused pipeline code

zivkovicmilos committed:
Add temporary node glue

zivkovicmilos committed:
Remove old server implementation

zivkovicmilos committed:
Update setup

zivkovicmilos committed:
Expand RPC request / response ID

zivkovicmilos committed:
Fixup the RPC client

zivkovicmilos committed:
Update listen address

zivkovicmilos committed:
Add leftover print fix comment

jaekwon committed:
Create tictac.md

Do not remove this file.
It is important information, and a canary file.

jaekwon committed:
Added file that proves Reddit censors truthful physics discoveries.

Of all places, /r/UFOResearch should not censor this kind of information.
DO NOT DELETE THIS FILE, DO NOT TRUST ANYONE WHO DELETES THIS FILE.

jaekwon committed:
Update tictac.md

added law of spinning mass.

moul committed:
fix(params): SetBytes should not alter the value bytes (alternative) (#4926)

Alternative for #4905 

Kept:
- Removed GetRaw/SetRaw since they were unused.
- SetBytes to use directly the store, preventing the value to be
aminoJSON encoded.

Updated:
- Preserve API consistency.

---------

Signed-off-by: moul <94029+moul@users.noreply.github.com>
Co-authored-by: Thomas Bruyelle <thomas.bruyelle@tendermint.com>

jeronimoalbi committed:
feat(examples): simple Markdown forms package (#4977)

Package aims to be a very simple package to define Gno-Flavored Markdown
forms.

It should be easy to update when new form elements or attributes are
introduced.

---------

Co-authored-by: Antoine Eddi <5222525+aeddi@users.noreply.github.com>

thehowl committed:
fix(vm/keeper): pre-load stdlib type check cache (#4925)

longer term fix for #4874, re-introducing the type check cache for
stdlibs.

jeronimoalbi committed:
feat(boards2): change realm to use `gno.land/p/gnoland/boards` (#4939)

PR changes Boards2 realm to use `gno.land/p/gnoland/boards` package
which makes the realm leaner.

It also improves `BasicPermissions` implementation by using a custom
members storage implementation that makes users iteration cheaper.

mvallenet committed:
fix(vm/keeper): improve error for non crossing msg call error (#4899)

fix #4432

---------

Co-authored-by: Morgan <morgan@morganbaz.com>

moul committed:
test: add a banker security txtar (#4987)

Signed-off-by: moul <94029+moul@users.noreply.github.com>

---------

Signed-off-by: moul <94029+moul@users.noreply.github.com>
Co-authored-by: Antoine Eddi <5222525+aeddi@users.noreply.github.com>

jeronimoalbi committed:
feat(boards2): open boards support (#4993)

jeronimoalbi committed:
feat(boards2): support Markdown forms (#4996)

jeronimoalbi committed:
chore(boards2): enable open boards field in create board form (#4998)

Also add a couple of minor fixes

jaekwon committed:
first commit of whitepaper; draft; DO NOT SHARE, deadman switch

moonia committed:
fix(gnoweb): handle multiline value for gno-textarea in forms (#4978)

Closes https://github.com/gnolang/gno/issues/4968

## Description

The `<gno-textarea>` component was literally displaying the `\n`
sequence. After fix, the parser now interprets the `\n` sequence as a
real newline character.

### Example

```
<gno-form>
  <gno-textarea name="content" value="         First line\nSecond line" />
</gno-form>
```

### Before fix: 
<img width="931" height="236" alt="Capture d’écran 2025-12-15 à 23 22
23"
src="https://github.com/user-attachments/assets/220f85ee-c5ad-4dd3-9b26-863bd84e6a58"
/>


### After fix: 
<img width="945" height="239" alt="Capture d’écran 2025-12-15 à 23 34
27"
src="https://github.com/user-attachments/assets/e8f7551e-749f-4259-ba50-5dae333cf487"
/>

---------

Co-authored-by: alexiscolin <alexis@jaunebleu.co>

moonia committed:
feat(gnoweb): support required attribute for markdown form fields (#4974)

## Description 
Closes https://github.com/gnolang/gno/issues/4970
This PR adds support for the required="true" attribute in Gno-flavored
markdown form fields.

### Usage 

- `required="true"` can be added to: `<gno-input>`, `<gno-textarea>`,
`<gno-select>`
- Required fields: display a * indicator next to the label

```
<gno-form>
    <gno-textarea name="bio" required="true" />
</gno-form>
```
### Preview
<img width="936" height="238" alt="Capture d’écran 2025-12-15 à 16 09
19"
src="https://github.com/user-attachments/assets/3cd743a8-350c-4906-b33f-25fb5ecb9ac8"
/>

---------

Co-authored-by: Alexis Colin <alexis@jaunebleu.co>

thehowl committed:
feat(github-bot): add MikaelVallenet to review team (#5036)

davd-gzl committed:
feat(valopers): add server type support for validator (#5015)

Co-authored-by: Miloš Živković <milos.zivkovic@tendermint.com>

mvallenet committed:
docs(stdlibs): fix outdated reference to the testing package (#4929)

Co-authored-by: Miloš Živković <milos.zivkovic@tendermint.com>

alexiscolin committed:
fix(boards2): improve ux / ui (#5022)

This PR enhances the boards2 realm UI with improved navigation, cleaner
post layout, and better visual hierarchy.

Related to https://github.com/gnolang/gno/issues/4999

## Changes
### Navigation Improvements

- Clickable breadcrumbs: "Boards" in headers now links back to the
boards list
- Back links on forms: Added "← Back to thread" and "← Back to board"
links on all form pages (Create Thread, Edit Thread, Repost Thread, Edit
Comment, Reply, Flag)
- Thread page breadcrumb: Shows [Boards] › [BoardName] with both
elements clickable to navigate into the realm from the realm

### Post Layout
- Author info at top: Moved author name and date to the beginning of
posts instead of the end to improve readability
- Format: **@user** · date #postID appears before the content
- Action links with arrow: Added ↳ prefix before action links (Flag •
Reply • Edit • Delete) to make it more obvious (as set of actions)
- Add a reply and repost counter to quickly identify popular posts and
trends.
- Remove the sidebar when the table of contents is empty.
- Reduce global spacing in the UI.

### Visual Hierarchy
- H6 headings for listings: Board names and thread titles in listings
now use H6 instead of bold text to add content to the sidebar.
- Roles badges have been added to enhance transparency and help users
identify who is speaking when messages pertain to rules.

### Forms
- Implements the `required` attributes on form fields to submit or
display an error when fields are empty or overlooked.
- A `Back to Previous Page` button is placed under the form to help
users understand that they can return after completing the form.

## Screenshots

<img width="1463" height="585" alt="Capture d’écran 2026-01-07 à 16 50
36"
src="https://github.com/user-attachments/assets/47477e49-45b4-4bc1-82b9-7751daa0228d"
/>
<img width="1464" height="982" alt="Capture d’écran 2026-01-07 à 16 50
55"
src="https://github.com/user-attachments/assets/04deb551-0472-43f9-ab1d-bb068369028b"
/>
<img width="1532" height="875" alt="Capture d’écran 2026-01-07 à 17 47
36"
src="https://github.com/user-attachments/assets/33af19c5-1644-4d1d-971b-9119eac48ea5"
/>

<img width="1386" height="445" alt="Capture d’écran 2026-01-08 à 21 42
13"
src="https://github.com/user-attachments/assets/f9aea63d-26b2-4687-9f8e-80fb105fdf31"
/>

---------

Co-authored-by: Jerónimo Albi <894299+jeronimoalbi@users.noreply.github.com>

gfanton committed:
feat(gnoweb): add `params:changed` (#5002)

This PR adds a new `params:changed` custom event to gnoweb, enabling
secure parameter update.

cc @jinoosss

---------

Co-authored-by: Alexis Colin <alexis@jaunebleu.co>

sw360cab committed:
chore(staging): staging env infra setup improvements (#5034)

Improved these items:
- Traefik config to get more precise data in logs
- Traefik config to persist Letsencrypt folder and avoid regenerating
each restart
- Gnoweb performances by adding Rate Limiter Traefik Middleware

moonia committed:
docs: fix typo errors across documentation (#5012)

moonia committed:
fix(docs): align docs with testnet test10 (#5021)

Co-authored-by: Nemanja Aleksic <kouteki@gmail.com>

zivkovicmilos committed:
feat: add initial `test10` deployment (#5045)

## Description

This PR adds the test10 deployment to the monorepo.

mvallenet committed:
feat(tm2): improve error message on invalid coins (#4942)

fix #4930 

I know TM2 is a "foundational" component we avoid to change.
I met a problem with not explicit error i tried to fix it, i could fix
it in the IssueCoin fn here but felt like it make more sense in tm2.

```go
// IssueCoin implements the Banker interface.
func (tb *TestBanker) IssueCoin(addr crypto.Bech32Address, denom string, amt int64) {
	coins := tb.CoinTable[addr]
	sum := coins.Add(tm2std.Coins{{Denom: denom, Amount: amt}})
	tb.CoinTable[addr] = sum
}
```

I was also thinking of improving this error message, but to keep it
scoped i did not do it in this PR

```go
func (e InvalidCoinsError) Error() string       { return "invalid coins error" }
```

This error is used why coins are not valid, but it does not explain what
is a valid coin (sorted, positive amount, lowercase denom)

---------

Co-authored-by: Antoine Eddi <5222525+aeddi@users.noreply.github.com>

thehowl committed:
fix(gnovm): allow divisions by zero on floats (#4961)

Turns out floats division by zero wasn't working properly; oops.

This PR makes sure we always put the result of floating point divisions
even when the divisor is zero.

zivkovicmilos committed:
Update params tests

zivkovicmilos committed:
Fixup block return types

zivkovicmilos committed:
Fixup genesis return types

zivkovicmilos committed:
Fixup status return types

zivkovicmilos committed:
Fixup net return types

zivkovicmilos committed:
Fixup consensus return types

zivkovicmilos committed:
Fixup mempool return types

zivkovicmilos committed:
Fixup tx return types

zivkovicmilos committed:
Fixup abci return types

zivkovicmilos committed:
docs: add multisig guide (#5072)

## Description

This PR adds a small guide to the docs, for `gnokey multisign` usage. It
also adds a small section on moving building / moving gnokey to a secure
machine.

---------

Co-authored-by: Antoine Eddi <5222525+aeddi@users.noreply.github.com>

zivkovicmilos committed:
Merge branch 'master' into dev/zivkovicmilos/rpc-server

jefft0 committed:
fix: tm2/Makefile: _test.flappy should set STABILITY_FILTER (#5047)

`make test` in the tm2 folder is supposed to run flappy tests with
[`_test.flappy`](https://github.com/gnolang/gno/blob/1183ce490502ce1e8e5d7eabf889994fbea3e932/tm2/Makefile#L47).
It sets
[`TEST_STABILITY=flappy`](https://github.com/gnolang/gno/blob/1183ce490502ce1e8e5d7eabf889994fbea3e932/tm2/Makefile#L52)
. This doesn't work because `FilterStability` expects
[`STABILITY_FILTER`](https://github.com/gnolang/gno/blob/1183ce490502ce1e8e5d7eabf889994fbea3e932/tm2/pkg/testutils/flappy.go#L16).
This changes the `Makefile` to set `STABILITY_FILTER`.

The CI checks don't run flappy tests. But now, running `make test`
locally may fail. For example,
[`TestFlappyBadBlockStopsPeer`](https://github.com/gnolang/gno/blob/1183ce490502ce1e8e5d7eabf889994fbea3e932/tm2/pkg/bft/blockchain/reactor_test.go#L198).

Related to "Deflap flappy tests" #202

---------

Signed-off-by: Jeff Thompson <jeff@thefirst.org>

alexiscolin committed:
fix(gnoweb): update ts URL construction in _fetchQEval to handle existing http protocols (#4964)

Added `normalizeRemoteURL()` function in main.go to ensure the remote
URL has a valid HTTP(S) protocol before being used by the frontend and
CSP.

- `tcp://` is converted to `http://` (RPC uses HTTP over TCP)
- Missing protocol defaults to `http://` (local)
- `http://` and `https:// `are kept as-is
- Updated `SecureHeadersMiddleware` to use the normalized URL for CSP
`connect-src`
- This prevents malformed URLs (e.g., http://https://...) and allows
users to use `-remote tcp://...` seamlessly.

Replaced Execute `<a>` link with `<button type="submit">` for proper
form submission. Updated form to use `method="GET"` with unique `id`
based on function name. Optimized JavaScript to update form `action`
attribute once and apply to both form and copy button, with proper URL
encoding to prevent parameter injection.

This PR addresses the issues with the default "true" value in radio
buttons and checkboxes while enhancing JavaScript performance by
avoiding multiple DOM fetches.

_Note: The root cause may be a misconfigured remote value in production
(e.g., missing the `:` in `https://`). This fix handles the symptom, but
the remote URL configuration might also be verified just in case._

_Note 2: The `.send` function query is not used in the submit button for
security reasons.._

zivkovicmilos committed:
Update gnodev proxy, support protocol:addr parsing

zivkovicmilos committed:
Remove noisy print

zivkovicmilos committed:
Remove redundant comments

zivkovicmilos committed:
Add server/client lib E2E tests

zivkovicmilos committed:
Add RPC E2E tests

zivkovicmilos committed:
Expose the node's RPC address for fetching

zivkovicmilos committed:
Update RPC client imports for gnodev

zivkovicmilos committed:
Revert "Expose the node's RPC address for fetching"

This reverts commit abf95fb66713ef553c4b635d335d7766783dc905.

zivkovicmilos committed:
Temporary save

zivkovicmilos committed:
Update node's RPC exposure

zivkovicmilos committed:
Update OnStop

jefft0 committed:
chore: Add genproto.yml (#4989)

Resolves part of issue https://github.com/gnolang/gno/issues/2025 .
* Remove unused `.pb.go` files. The output of protoc will be kept in a
different repo. ([gno-types](https://github.com/gnolang/gno-types)?)
Keep the `.proto` files.
* Add CI job `genproto.yml` to run `make -C misc/genproto` and fail if
there are differences
* We haven't run this in a while, so run it and commit the changes to
`.proto` files

BREAKING CHANGE: This changes many generated `.proto` files. Any
application which uses them may need to be updated.

---------

Signed-off-by: Jeff Thompson <jeff@thefirst.org>
Co-authored-by: aeddi <antoine.e.b@gmail.com>
Co-authored-by: Antoine Eddi <5222525+aeddi@users.noreply.github.com>

alexiscolin committed:
docs: fix inline code blocks spanning multiple lines (#5100)

This PR fixes formatting issues where inline code blocks (single
backticks) span across multiple lines in the documentation. Inline code
in Markdown using single backticks (`` `code` ``) should remain on a
single line. When line breaks appear inside inline code blocks, it can
cause:
- Rendering inconsistencies across different Markdown parsers
- Unexpected formatting in documentation sites
- Potential parsing errors in strict Markdown processors

**It causes MDX rendering
[issues](https://github.com/gnolang/docs.gno.land/actions/runs/21363843094/job/61490027463)
in our gno.docs documentation system.**

jeronimoalbi committed:
fix(govdao): extra argument checks and validations (#5078)

PR adds extra argument checks to validate some edge cases that were not
handled and updates the supermajority 2/3s threshold.

It also adds two security changes, one to make sure that allowed list
can't be changed from `Proposal` references, and another to explicitly
assert `init` realm public functions access.

moonia committed:
docs: fix effective-gno.md typos (#5041)

clarity and punctuation fixes.

jefft0 committed:
chore: TestHandshakeReplayNone is not flappy (#5050)

It seems that flappy tests have not been run for at least 3 years. (See
PR #5047.) One of the tests seems to not be flappy.

* Convert `TestHandshakeReplayNone` to a normal test. Remove
`FilterStability(t, testutils.Flappy)`
* In this PR, push an empty commit 10 times to repeatedly re-run the CI
checks. It passes each time.

<img width="455" height="393" alt="image"
src="https://github.com/user-attachments/assets/2f8ba6c0-976d-4241-8ed1-4b8cfea0083b"
/>

Also, on my local macOS I ran the test 100 times. It passed.
```
for run in {1..100}; do
  go test -v ./tm2/pkg/bft/consensus/... -run TestHandshakeReplayNone -count=1
done
```

The other flappy tests are still flappy, which we'll address in a
separate PR.

Related to #202

---------

Signed-off-by: Jeff Thompson <jeff@thefirst.org>
Co-authored-by: Manfred Touron <94029+moul@users.noreply.github.com>

aeddi committed:
fix(benchops): avoid panic by supporting recursive store timing (#4921)

ltzmaxwell committed:
chore(test): fix runtime error while update golden tests (#5102)

fix: #5101 .

mvallenet committed:
docs: transform relative path from outside docs folder into absolute path (#5025)

fix #5023

ltzmaxwell committed:
chore(gnovm): improve type check msg (#4936)

this PR improves the error message for comparison mismatch that
mentioned in #4911.

mvallenet committed:
chore: update deadlinks (#5013)

This PR fixes broken and outdated links throughout the codebase:
- Fix typo in CONTRIBUTING.md: Missing closing parenthesis in Awesome
Gno link
- Update Makefile commands in READMEs: install_gnokey → install.gnokey,
install_gno → install.gno
- Fix tx-archive path: Updated to
gnolang/gno/tree/master/contribs/tx-archive
- Fix Tendermint links: Updated tendermint/classic references to
tendermint/tendermint throughout tm2/
- Update documentation URLs: Migrated from deprecated
tendermint.com/docs/ to docs.tendermint.com/ and docs.cometbft.com/
- Fix GitHub path formats: Corrected malformed paths like
gnolang/gno/tm2/pkg/bft/ to proper gnolang/gno/tree/master/tm2/pkg/bft/
format
- Fix external references: Updated cosmos-sdk, iavl, and tmlibs links to
correct repositories
- Remove dead references: Removed broken links to non-existent files
(e.g., zrealm_nft3.gno, outdated ABCI docs)

jaekwon committed:
fix(gnovm): uverse didupdate issue (#5019)

fixed a security issue that wasn't caught by Oak introduced by deelawn.
I think it's good but some tests are failing so needs further
inspection.

---------

Co-authored-by: Manfred Touron <94029+moul@users.noreply.github.com>
Co-authored-by: Morgan Bazalgette <morgan@morganbaz.com>
Co-authored-by: ltzMaxwell <ltz.maxwell@gmail.com>

moonia committed:
fix(docs): improve clarity in gno-testing guide (#5042)

Improve wording + fix minor grammar issues.

omarsy committed:
refactor(gnovm): not recursive save when object is dirty (#4938)

alternative #4922

fix #4818

---------

Co-authored-by: Jae Kwon <53785+jaekwon@users.noreply.github.com>
Co-authored-by: ltzmaxwell <ltz.maxwell@gmail.com>
Co-authored-by: jaekwon <jae@tendermint.com>

alexiscolin committed:
fix(gnoweb): md form path validation (#5046)

omarsy committed:
fix(gnovm/preprocess): forbid range on invalid values (#5017)

closes #4680

---------

Co-authored-by: Morgan Bazalgette <morgan@morganbaz.com>

zivkovicmilos committed:
Update gnodev RPC usage

thehowl committed:
fix(github-bot): remove rule for docs directory (#5105)

There are no longer properly "devrels" in the team; the PR's relating to
docs can just follow the normal approval process without this rule.

zivkovicmilos committed:
Add cleanup for OnStop

zivkovicmilos committed:
Merge branch 'master' into dev/zivkovicmilos/rpc-server

zivkovicmilos committed:
Fix tx-archive imports

zivkovicmilos committed:
Fixup type returns

zivkovicmilos committed:
Make GitHub security scanning happy

zivkovicmilos commented on tm2/pkg/bft/node/node.go:
```diff
@@ -573,24 +571,31 @@ func (n *Node) OnStart() error {
-		listeners, err := n.startRPC()
-		if err != nil {
-			return err
+		// Initialize the JSON-RPC pipeline
```
I think we can improve this, I don't particularly like this heavy setup code, maybe we can move it out somewhere else

mvallenet reviewed (commented):
FYI: I did not finish review, it's dense and i'll do it a little bit everyday.
I ask some questions i could potentially find answer myself but i'm interested in your thoughts and it will help me go deeper in the review.

thehowl commented:
Closing as very stale; if someone picks it up it needs plenty of work anyway.
